react/Stack/styles.styl

```diff
@@ -1,5 +1,9 @@
 @require 'utilities/spaces.styl'
 
+.Stack
+    display flex
+    flex-direction column
+
```

Comment:
Switching to flexbox has some side effects that I'm not sure we want. If I'm not mistaken, elements will now `stretch` by default on the width (https://developer.mozilla.org/en-US/docs/Web/CSS/align-items), and all elements will be allowed to shrink.

IMO it's a bit risky, it can work if we set all the properties right but maybe applying a `display inline-block` on the `inline` elements is a better solution?

Reply:
We can see via Argos that in the current state it effectively has undesirable effects (on CollectionField). I think I'm going to split the second commit in its own PR since it's the problematic one.

Reply:
See https://github.com/cozy/cozy-ui/pull/1301